client/src/features/posts/AddPost.js

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addNewPost } from './postsThunks';
+
+const AddPostForm = ({ history }) => {
+    const dispatch = useDispatch();
+    const [title, setTitle] = useState('');
+    const [text, setText] = useState('');
+    const [categories, setCategories] = useState([]);
+
+    const handleAddPost = async () => {
+        try {
+            const newPost = {
+                title,
+                text,
+                categories: categories.split(',').map((category) => category.trim()),
+                // Add other fields as needed based on your server model
```

Comment:
![image](https://github.com/kevinleey/SeniorForum/assets/55768372/1ad0a7e2-70e1-449f-ac32-0fa9e95e9183)

This is currently what a post looks like on our database, so we will need some additional information/fields when making a POST call to update the DB. I think you should be able to find a way to pass a user ObjectID in the `postedBy` field, a new date in the date field and a blank array for `comments`.

Reply:
I think the user ID will be tracked, I am still working on implementing it. Also the dateCreated will be logged automatically when they hit add post. (I am also still working on this)